plugins/feeds/public/otx_alienvault.py

```diff
@@ -0,0 +1,211 @@
+
+        OTXAlienvault.__create_list_observables(
+            Url,
+            'URL',
+            item['indicators'],
+            observables,
+            'urls')
+
+        OTXAlienvault.__create_list_observables(
+            Hostname,
+            'domain',
+            item['indicators'],
+            observables,
+            'domains')
+
+        OTXAlienvault.__create_list_exploit(
+            Exploit,
+            'CVE',
+            item['indicators'],
+            observables,
+            'exploits')
+
+        OTXAlienvault.__create_list_observables(
+            Hash,
+            'FileHash-SHA256',
+            item['indicators'],
+            observables,
+            'sha256')
+
+        OTXAlienvault.__create_list_observables(
+            Hash,
+            'FileHash-MD5',
+            item['indicators'],
+            observables,
+            'md5')
+
+        OTXAlienvault.__create_list_observables(
+            Hash,
+            'FileHash-SHA1',
+            item['indicators'],
+            observables,
+            'sha1')
+
+        tags = item['tags']
+
+        context['references'] = '\r\n'.join(item['references'])
+        context['description'] = item['description']
+        context['link'] = 'https://otx.alienvault.com/pulse/%s' % item['id']
+
+        OTXAlienvault.__add_contex(context, observables)
+        OTXAlienvault.__add_source(observables)
+        if tags:
+            OTXAlienvault.__add_tags(tags, observables)
+
+        if 'sha256' in observables and observables['sha256']:
+            self.__create_link_hashes_and_network_indic(
+                observables['sha256'],
+                observables['urls'])
+            self.__create_link_hashes_and_network_indic(
+                observables['sha256'],
+                observables['domains'])
+            self.__create_link_hashes_and_network_indic(
+                observables['sha256'],
+                observables['hostnames'])
+
+        if 'sha1' in observables and observables['sha1']:
+            self.__create_link_hashes_and_network_indic(
+                observables['sha1'],
+                observables['urls'])
+
+            self.__create_link_hashes_and_network_indic(
+                observables['sha1'],
+                observables['domains'])
+
+            self.__create_link_hashes_and_network_indic(
+                observables['sha1'],
+                observables['hostnames'])
+
+        if 'md5' in observables and observables['md5']:
+            self.__create_link_hashes_and_network_indic(
+                observables['md5'],
+                observables['urls'])
+
+            self.__create_link_hashes_and_network_indic(
+                observables['md5'],
+                observables['domains'])
+
+            self.__create_link_hashes_and_network_indic(
+                observables['md5'],
+                observables['hostnames'])
+
+        if ('hostnames' in observables and observables['hostnames']) or (
```

Comment:
Isn't this what you want?

```suggestion
        if observables.get('hostnames') and observables.get('domains'):
```